### PR TITLE
⭐️ Add AKS tests

### PR DESCRIPTION
**Pull request by imilchev**
I still need to make sure we have the azure secrets defined in the repo before merging this

### CODE REVIEW

File: .github/terraform/aks/main.tf [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  keepers = {
+    # Generate a new ID only when a new resource group is defined
+    resource_group = azurerm_resource_group.rg.name
+  }
+
+  byte_length = 8
+}
+
+# Create storage account for boot diagnostics
+resource "azurerm_storage_account" "mystorageaccount" {
```
Comment (czunker) [inline .github/terraform/aks/main.tf:23]: Do we really need this for the tests?

File: .github/terraform/aks/main.tf [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  name                     = "diaglunalectric${random_string.suffix.result}"
+  location                 = azurerm_resource_group.rg.location
+  resource_group_name      = azurerm_resource_group.rg.name
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+# Create (and display) an SSH key
+resource "tls_private_key" "attacker_vm_ssh" {
```
Comment (czunker) [inline .github/terraform/aks/main.tf:32]: I think this can be removed.

File: .github/terraform/aks/main.tf [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  depends_on = [azurerm_kubernetes_cluster.cluster]
+}
+
+resource "azurerm_key_vault_secret" "example-secret" {
+  name         = "secret-sauce"
+  value        = "example-pass"
+  key_vault_id = azurerm_key_vault.keyvault.id
+}
+
+resource "azurerm_key_vault_secret" "ssh-key" {
```
Comment (czunker) [inline .github/terraform/aks/main.tf:124]: I think this can be removed.

File: .github/terraform/aks/main.tf [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      "Create",
+      "Get",
+      "List",
+    ]
+  }
+
+  depends_on = [azurerm_kubernetes_cluster.cluster]
+}
+
+resource "azurerm_key_vault_secret" "example-secret" {
```
Comment (czunker) [inline .github/terraform/aks/main.tf:118]: Do we need this?

File: .github/terraform/aks/main.tf [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+  tags = {
+    Environment = "Mondoo Operator Tests"
+  }
+}
+
+# configure keyvault
+data "azurerm_client_config" "current"{}
+
+resource "azurerm_key_vault" "keyvault" {
```
Comment (czunker) [inline .github/terraform/aks/main.tf:75]: I see it is part of the node pool tags, but what is it good for?

File: .github/terraform/aks/output.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+output "resource_group_name" {
+  value = azurerm_resource_group.rg.name
+}
+
+output "tls_private_key" {
```
Comment (czunker) [inline .github/terraform/aks/output.tf:5]: I think this can be removed.

File: .github/workflows/cloud-tests.yaml [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+            .github/terraform/aks/aks-${{ matrix.k8s-version }}.json
+
+      - name: Upload test logs artifact
+        uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: test-logs-aks-${{ matrix.k8s-version }}
+          path: /home/runner/work/mondoo-operator/mondoo-operator/tests/integration/_output/
+
+  # release-tests-eks:
```
Comment (czunker) [inline .github/workflows/cloud-tests.yaml:107]: I think this can be removed. I will add my the EKS tests here after I re-based.

**Comment by imilchev** [comment]
I cleaned up all the unnecessary stuff. When running the terraform template locally it seems to work fine

**Pull request merged**